sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -310,6 +311,9 @@ class RocksDB(
         "checkpoint" -> checkpointTimeMs,
         "fileSync" -> fileSyncTimeMs
       )
+      // reset resources as we already pushed the changes and it has been committed
```

Comment:
Should we do this in the finally clause to prevent any exception/error causing this not to run?

Reply:
Actually I intended to put this here, as I thought we have to clean up only when it is successfully committed. But looks like it's also safe to place this to `finally`, as we invalidate the loaded version in any way if any exception happens. Will do.